### ISSUE TITLE
As a developer, I would like an ingress to access the vault ui and configure secrets

### ISSUE THREAD

**Issue by SonOfLope** [ai-cfia/howard#56]
In order to allow our developers to manage their secrets, we will need to configure an ingress to access the vault ui.

This will be done after an authentication solution has been implemented for the cluster.

- [x] Configure vault ingress to access the UI
- [x] Configure Github authentication with pat tokens for admin users
- [x] Configure auth for kubernetes Service Account to be able to access vault
- [x] Configure Policies and roles for kubernetes to read/write to secret engine
- [x] Configure Automated configuration of policies, roles and secret engine. (currently manually created) (This will be done in https://github.com/ai-cfia/howard/issues/99)
- [x] Add documentation and trainings on how to use vault

**Comment by SonOfLope** [comment]
This ingress was tested locally and proven to work (will be implemented once the cluster is secure): 
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: vault-ui
  namespace: vault
  labels:
    app.kubernetes.io/name: vault-ui
    app.kubernetes.io/instance: vault
  annotations:
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
    cert-manager.io/issuer: "letsencrypt-prod"
    ingress.kubernetes.io/force-ssl-redirect: "true"
    kubernetes.io/tls-acme: "true"
spec:
  ingressClassName: nginx
  tls:
    - hosts:
        - "vault.azure.sonoflope.ca"
      secretName: vault-ui
  rules:
    - host: "vault.azure.sonoflope.ca"
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: vault-ui
                port:
                  name: https

**Comment by SonOfLope** [comment]
We got a working ingress with : https://vault.inspection.alpha.canada.ca. Only thing left is configure authentication for kubernetes to read and write to vault.

**Comment by SonOfLope** [comment]
Vault is currently configured to work with kubernetes and manage secrets. It is working as expected. I set the issue as block since some configuration were made manually and needs to be automated. These changes require some research. As twe have other priorities, I will work on this later this week. 

TLDR : Vault is working and we can deploy apps needing secret management

**Comment by ThomasCardin** [comment]
Documentation + training material should also be part of the list of tasks.

**Comment by ThomasCardin** [comment]
This will be done in https://github.com/ai-cfia/howard/issues/99
